src/graphics/vertex_buffer.rs

```diff
@@ -0,0 +1,246 @@
+use crate::graphics::csfml_graphics_sys::*;
+use crate::graphics::{Drawable, PrimitiveType, RenderStates, RenderTarget, Vertex};
+
+/// Usage specifiers for a [`VertexBuffer`]
+///
+/// If data is going to be updated once or more every frame,
+/// set the usage to `Stream`.  If data is going to be set once and used
+/// for a long time without being modified, set the usage to `Static`.
+/// For everything else `Dynamic` should be a good compromise.
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum VertexBufferUsage {
+    /// Constantly changing data.
+    Stream = 0,
+    /// Occasionally changing data.
+    Dynamic = 1,
+    /// Rarely changing data.
+    Static = 2,
+}
+
+impl VertexBufferUsage {
+    pub(super) fn raw(self) -> sfVertexBufferUsage {
+        unsafe { ::std::mem::transmute(self) }
```

Comment:
I'm sorry, this is nitpicking, but can't we do a simple `as` conversion here instead of unsafe `transmute`?
I know it shouldn't technically change anything, but unnecessary unsafe always bothers me. We should strive to avoid unsafe whenever possible.

Reply:
Nitpicking is a good habit :)
I removed the unnecessary unsafe code.

Reply:
Btw, isn't the same objection valid for all the other enums using `transmute` in their `raw` method? (e.g. `PrimitiveType`, `mouse::Button`, etc)

Reply:
> Btw, isn't the same objection valid for all the other enums using `transmute` in their `raw` method? (e.g. `PrimitiveType`, `mouse::Button`, etc)

That might be the case. If so, they should probably be changed to not use unsafe.

Reply:
Ok, I think I'll do another PR for that in the near future.